packages/react/src/_popover/Popover.tsx

```diff
@@ -79,43 +83,31 @@ export function usePopover<RT extends ReferenceType = ReferenceType>(
 			// Adds distance between the reference and floating element
 			// https://floating-ui.com/docs/offset
 			offset(offsetOption),
-			// Changes the placement of the floating element in order to keep it in view
-			// https://floating-ui.com/docs/flip
-			flip({ padding: DEFAULT_OFFSET }),
 			// Allows you to change the size of the floating element
 			// https://floating-ui.com/docs/size
+			shift({ padding: MIN_SIDE_GUTTER_WIDTH }),
 			size({
 				padding: DEFAULT_OFFSET, // Prevents the floating element hit the edge of the screen
-				apply({
-					availableWidth,
-					availableHeight: _availableHeight,
-					elements,
-					rects,
-				}) {
-					// Popovers can have a predefined max-height if there is enough room on the screen
-					const maxHeight =
-						maxHeightOption && _availableHeight > maxHeightOption
-							? maxHeightOption
-							: _availableHeight;
-
-					// Minimum acceptable height before `flip` will take over
-					const availableHeight = minHeightOption
-						? Math.max(minHeightOption, maxHeight)
-						: maxHeight;
-
+				apply({ elements, rects }) {
 					Object.assign(elements.floating.style, {
-						maxHeight: `${availableHeight}px`,
+						height: 'max-content',
+						...(maxHeight !== undefined && { maxHeight: `${maxHeight}px` }),
+						...(minHeight !== undefined && { minHeight: `${minHeight}px` }),
 						// https://floating-ui.com/docs/size#match-reference-width
 						...(matchReferenceWidth
 							? {
 									width: `${rects.reference.width}px`,
 							  }
 							: {
-									maxWidth: `${availableWidth}px`,
+									maxWidth: `calc(100vw - ${2 * MIN_SIDE_GUTTER_WIDTH}px)`,
+									overflowX: 'auto',
 							  }),
 					});
 				},
 			}),
+			// Changes the placement of the floating element in order to keep it in view
+			// https://floating-ui.com/docs/flip
+			flip({ padding: DEFAULT_OFFSET }),
```

Comment:
Not sure if there's anything we can do about this, but, if the calendar can never fit, i think positioning to the bottom so you can always see the top of the calendar is better than putting it above and chopping off the top 
<img width="390" alt="image" src="https://github.com/agriculturegovau/agds-next/assets/853552/5f79f0ba-a72a-4600-916c-9dddc9bd0cce">

Reply:
If the calendar cannot fit because the screen is too small, it should prioritise covering the input over hiding any part of the calendar.

If the screen is so small that the calendar gets cut off even if it is completely covering the screen then we should resort to a scrollable calendar.

Reply:
Although there are [fallback props with floating UI](https://floating-ui.com/docs/flip#fallbackplacements), they cause additional UI jumps for our DatePicker. On the balance of it, they aren't worthwhile, IMO, as they detract from the UX.